### ISSUE TITLE
Add proxy support

### ISSUE THREAD

**Issue by zhum** [protoncr/tourmaline#21]
Please, add proxy support (http and/or socks5).

**Comment by watzon** [comment]
This is one thing I do have planned, but it requires having proxy support in the HTTP client, which right now we don't have. I'll look into some solutions, but this is a good issue to have here anyway. Thanks :)

**Comment by watzon** [comment]
Waiting on icyleaf/halite#57 which it looks like hasn't had any movement for a while unfortunately

**Comment by igor-alexandrov** [comment]
Maybe consider moving to Crest? https://github.com/mamantoha/crest

**Comment by watzon** [comment]
I'm actually just using the standard HTTP::Client now because Halite was too slow. Someone could make an argument for Crest though if they added built in connection pooling and we could show a speed improvement, or at the very least no loss of speed.

**Comment by watzon** [comment]
For now I have managed to implement proxy support using [mamantoha/http_proxy](https://github.com/mamantoha/http_proxy). As of aa8e089 the `Tourmaline::Client` initializer now accepts several proxy related arguments. Has been tested and is working on my end, so I'm going to close this now. Open another issue if you have any problems!
